infra/testApi.mjs

```diff
@@ -3,177 +3,317 @@ import { strict as assert } from 'node:assert';  // use strict equality everywhe
-const jid = sampleRSP.headers.get("x-herbie-job-id")
-assert.notEqual(jid, null)
-const sample = await sampleRSP.json()
-assertIdAndPath(sampleAsyncResult)
-assert.ok(sampleAsyncResult.points)
-assert.equal(sampleAsyncResult.points.length, SAMPLE_SIZE)
-assertIdAndPath(sample)
-assert.ok(sample.points)
-assert.equal(sample.points.length, SAMPLE_SIZE, `sample size should be ${SAMPLE_SIZE}`)
-
-// Make second call to test that results are the same
-const sample2RPS = await fetch(makeEndpoint("/api/sample"), sampleBody)
-const jid2 = sample2RPS.headers.get("x-herbie-job-id")
-assert.notEqual(jid2, null)
-const sample2 = await sample2RPS.json()
-assertIdAndPath(sample2)
-assert.deepEqual(sample.points[1], sample2.points[1])
-
-//Explanations endpoint
-const explainBody = {
-  method: 'POST',
-  body: JSON.stringify({
-    formula: FPCoreFormula, sample: sample.points
-  })
+
+// MARK: Translate
+async function callTranslate(fpcore, language, maybe_value) {
+  if (language === 'mathjs') {
+    console.debug(`Translate to mathjs for ${fpcore}.`);
+    const mathjs_rsp = await fetchAndCheckRSPHeaders("/api/mathjs", {
+      method: 'POST', body: JSON.stringify({ formula: fpcore })
+    }, CHECK_CORS);
+    const mathjs = await mathjs_rsp.json();
+    if (maybe_value != null) {
+      assert.equal(mathjs.mathjs, maybe_value);
+    } else {
+      assert.equal(mathjs.mathjs.length > 0, true);
+    }
+  } else {
+    console.debug(`Translate to ${language} for ${fpcore}.`);
+    const json = await (await fetchAndCheckRSPHeaders("/api/translate", {
+      method: 'POST', body: JSON.stringify(
+        { formula: fpcore, language: language })
+    }, CHECK_CORS)).json();
+    assert.equal(Object.values(json).length, 2);
+    assert.equal(json['language'], language);
+    if (maybe_value != null) {
+      assert.equal(json['result'], maybe_value);
+    } else {
+      assert.equal(json['language'].length > 0, true);
+    }
+  }
 }
-const explain = await (await fetch(makeEndpoint("/api/explanations"), explainBody)).json()
-assertIdAndPath(explain)
-assert.equal(explain.explanation.length > 0, true, 'explanation should not be empty');
-const explainAsyncResult = await callAsyncAndWaitJSONResult("/api/start/explanations", explainBody)
-assertIdAndPath(explainAsyncResult)
-assert.equal(explainAsyncResult.explanation.length > 0, true, 'explanation should not be empty');
-
-// Analyze endpoint
-const errorsBody = {
-  method: 'POST', body: JSON.stringify({
-    formula: FPCoreFormula, sample: [[[
-      14.97651307489794
-    ], 0.12711304680349078]]
-  })
+
+// Test with a likely missing job-id
+const badTimelineRSP = await fetchAndCheckRSPHeaders("/timeline/42069", { method: 'GET' }, CHECK_CORS)
+assert.equal(badTimelineRSP.status, 404);
+
+// MARK: Check Status
+const check_missing_job = await fetchAndCheckRSPHeaders(`/check-status/42069`, { method: 'GET' }, CHECK_CORS)
+assert.equal(check_missing_job.status, 202);
```

Comment:
If this is the current behavior I guess that's fine but why does this return 202? Shouldn't this return an error response?

Reply:
Ya, this is the current behavior. Follow up PR changing this to return a 404?

Reply:
Sure, separate PR.